docs/src/main/sphinx/functions/json.rst

```diff
@@ -5,13 +5,38 @@ JSON functions and operators
 Cast to JSON
 ------------
 
-Casting from ``BOOLEAN``, ``TINYINT``, ``SMALLINT``, ``INTEGER``,
-``BIGINT``, ``REAL``, ``DOUBLE`` or ``VARCHAR`` is supported.
-Casting from ``ARRAY``, ``MAP`` or ``ROW`` is supported when the element type of
-the array is one of the supported types, or when the key type of the map
-is ``VARCHAR`` and value type of the map is one of the supported types,
-or when every field type of the row is one of the supported types.
-Behaviors of the casts are shown with the examples below::
+The following types can be cast to JSON:
+
+* ``BOOLEAN``
+* ``TINYINT``
+* ``SMALLINT``
+* ``INTEGER``
+* ``BIGINT``
+* ``REAL``
+* ``DOUBLE``
+* ``VARCHAR``
+
+Additionally, ``ARRAY``, ``MAP``, and ``ROW`` types can be cast to JSON when
+the following requirements are met:
+
+* ``ARRAY`` types can be cast when the element type of the array is one
+    of the supported types.
```

Comment:
```suggestion
    of the supported types listed above.
```

Reply:
Google style guide advises against referring to other sections of a doc with directions.